source/agent/conference/conference.js

```diff
@@ -778,9 +780,43 @@ var Conference = function (rpcClient, selfRpcId) {
         log.warn('Add duplicate subscription:', id);
       }
     }
+    const switchMap = new Map();
     for (const from of subscription.froms()) {
       const streamId = streams[from] ? from : trackOwners[from];
       if (streams[streamId]) {
+        if (streams[streamId].type === 'forward' && room_config.enableBandwidthAdaption) {
+          // Create layer stream or find simulcast tracks
+          const svcTrack = streams[streamId].media.tracks.find(t => t.scalabilityMode);
+          let createSwitch = null;
+          if (svcTrack) {
+            // Make RPC to create layer streams
+            const layerOption = svcTrack.source !== 'screen-cast' ?
+              {spatial: true, temporal: false} :
```

Comment:
can you explain why you treat screen-cast differently?

Reply:
screen sharing like ppt often requires high resolution